src/Adapter/News/NewsDataProvider.php

```diff
@@ -0,0 +1,173 @@
+ */
+class NewsDataProvider
+{
+    const NUM_ARTICLES = 2;
+
+    const CLOSED_ALLOWED_FAILURES = 3;
+    const CLOSED_TIMEOUT_SECONDS = 3;
+
+    const OPEN_ALLOWED_FAILURES = 3;
+    const OPEN_TIMEOUT_SECONDS = 3;
+    const OPEN_THRESHOLD_SECONDS = 86400; // 24 hours
+
+    /**
+     * @var CircuitBreakerInterface
+     */
+    private $circuitBreaker;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var int
+     */
+    private $contextMode;
+
+    /**
+     * @var CountryDataProvider
+     */
+    private $countryDataProvider;
+
+    /**
+     * @var Tools
+     */
+    private $tools;
+
+    /**
+     * @var Validate
+     */
+    private $validate;
+
+    /**
+     * NewsDataProvider constructor.
+     *
+     * @param CircuitBreakerInterface $circuitBreaker
+     * @param CountryDataProvider $countryDataProvider
+     * @param Tools $tools
+     * @param Configuration $configuration
+     * @param Validate $validate
+     * @param int $contextMode
+     */
+    public function __construct(
+        CircuitBreakerInterface $circuitBreaker,
+        CountryDataProvider $countryDataProvider,
+        Tools $tools,
+        Configuration $configuration,
+        Validate $validate,
+        $contextMode
+    ) {
+        $this->circuitBreaker = $circuitBreaker;
+        $this->configuration = $configuration;
+        $this->contextMode = $contextMode;
+        $this->countryDataProvider = $countryDataProvider;
+        $this->tools = $tools;
+        $this->validate = $validate;
+    }
+
+    /**
+     * @param string $isoCode
+     *
+     * @return array
+     *
+     * @throws \PrestaShopException
+     */
+    public function getData($isoCode)
+    {
+        $data = ['has_errors' => true, 'rss' => []];
+        $apiUrl = $this->configuration->get('_PS_API_URL_');
+
+        $blogXMLResponse = $this->circuitBreaker->call($apiUrl . '/rss/blog/blog-' . $isoCode . '.xml');
+
+        if (empty($blogXMLResponse)) {
+            return $data;
+        }
+
+        $rss = @simplexml_load_string($blogXMLResponse);
+        if (!$rss) {
+            return $data;
+        }
+
+        $articles_limit = self::NUM_ARTICLES;
+
+        $shop_default_country_id = (int) $this->configuration->get('PS_COUNTRY_DEFAULT');
+        $shop_default_iso_country = mb_strtoupper($this->countryDataProvider->getIsoCodebyId($shop_default_country_id), 'utf-8');
+        $analytics_params = [
+            'utm_source' => 'back-office',
+            'utm_medium' => 'rss',
+            'utm_campaign' => 'back-office-' . $shop_default_iso_country,
+        ];
+
+        foreach ($rss->channel->item as $item) {
+            if ($articles_limit == 0
+                || !$this->validate->isCleanHtml((string) $item->title)
+                || !$this->validate->isCleanHtml((string) $item->description)
+                || !isset($item->link, $item->title)) {
+                break;
```

Comment:
Careful, you use break for each conditions. The `break` makes sense when you reach the limit, but maybe in the other cases you should use `continue` or else any invalid item will stop the loop and you can't reach the expected limit.

Reply:
also, maybe you could use `empty($item->link, $item->title)` instead of `!isset`

Reply:
@jolelievre So stupid I was. Thanks.